source/common/event/timer_impl.h

```diff
@@ -16,7 +16,33 @@ namespace Event {
  */
 class TimerUtils {
 public:
-  static void millisecondsToTimeval(const std::chrono::milliseconds& d, timeval& tv);
+  /**
+   * Intended for consumption by enable(HR)Timer, this method is templated method to avoid implicit
+   * duration conversions for its input arguments. This lets us have an opportunity to check bounds
+   * before doing any conversions. When the passed in duration exceeds INT32_T max days, the output
```

Comment:
Nit: I think this should just say "INT32_MAX seconds", otherwise it is hard to interpret. Same below, the "INT32_MAX days worth of seconds" may be hard to understand.

Reply:
Sorry - that was a confusing comment indeed; amended: https://github.com/envoyproxy/envoy/pull/9229/commits/0c57cfcf5a2b2d2b04ddb5fae1ffe067ccb20cbb